gems/smithy-client/lib/smithy-client/protocols/rpc_v2.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Protocols
+      # A RPC-based protocol over HTTP that sends requests
+      # and responses with CBOR payloads.
+      #
+      # TODO: Refactor methods to handle eventstreams
+      class RPCv2
+        SHAPE_ID = 'smithy.protocols#rpcv2Cbor'
+
+        def initialize(options = {})
+          @query_compatible = options[:query_compatible]
+        end
+
+        # @api private
+        def build(context)
+          codec = Client::Codecs::CBOR.new(setting(context))
+          context.request.body = codec.serialize(context.params, context.operation.input)
+          context.request.http_method = 'POST'
```

Comment:
This also needs to set a path of `/service/{serviceName}/operation/{operationName}` on the endpoint, see: https://smithy.io/2.0/additional-specs/protocols/smithy-rpc-v2.html#requests

Reply:
Ahhhhh good catch, I was so focused on the build/parse that I didn't migrate those portions over. Let me double check!